sinks/graphite.py

```diff
@@ -142,16 +160,23 @@ def _write_metric(self, metric):
                              self.attempts)
 
 
-if __name__ == "__main__":
-    # Initialize the logger
-    logging.basicConfig()
+def main():
 
     # Intialize from our arguments
     graphite = GraphiteStore(*sys.argv[1:])
 
     # Get all the inputs
-    metrics = sys.stdin.read()
+    while True:
```

Comment:
If you read stdin like that your line splitting and iterating doesn't happen until after EOF is read. If you loop over raw_input however you can act on each line as it is read

```>>> import sys
>>> for line in sys.stdin:
...     print line
... 
foo
bar
bing
bong
foo

bar

bing

bong

^CTraceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyboardInterrupt

>>> 
>>> while True:
...     try:
...             print raw_input().strip()
...     except EOFError:
...             break
... 
foo
foo
bar
bar
bing
bing
bong
bong
```

Perhaps a trivial optimization, but my intent was to not hold all of stdin in memory before parsing. I suppose a change like this should be supported by some empirical evidence. I'll run a benchmark when I find a few extra minutes.